src/Provider/AbstractProvider.php

```diff
@@ -357,12 +379,17 @@ protected function fetchProviderData($url, array $headers = [])
             $response = (string) $httpResponse->getBody();
         } catch (HttpAdapterException $e) {
             // @codeCoverageIgnoreStart
-            $raw_response = explode("\n", (string) $e->getResponse()->getBody());
-            throw new IDPException(end($raw_response));
+            $response = (string) $e->getResponse()->getBody();
             // @codeCoverageIgnoreEnd
         }
 
-        return $response;
+        $result = $this->parseResponse($response);
```

Comment:
I agree. Let's do that. It doesn't make sense for the provider to have all these helper methods. Some providers will use them, others won't. We should leave it up to them to return a `UserInterface`, and they can extend it however it seems fit for the service they are using.